src/shared/helpers/react/getFieldWithComponent.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import { Field, WrappedFieldProps, BaseFieldProps } from 'redux-form';
+import { Omit, MergeRight } from '_helpers';
+
+function getFieldWithComponent<P extends WrappedFieldProps>(Component: React.ComponentType<P>) {
+  type OwnProps = Omit<P, keyof WrappedFieldProps>;
+  type FieldProps = Omit<BaseFieldProps<OwnProps>, 'component'>;
+  type ResultProps = MergeRight<OwnProps, FieldProps>;
+
+  const result: React.StatelessComponent<ResultProps> = (props: ResultProps) =>
+    <Field<OwnProps> {...props as any} component={Component} />;
+  result.displayName = `FieldWithComponent(${Component.displayName || Component.name || 'Component'})`;
+  return result;
+}
+
+export default getFieldWithComponent;
```

Comment:
Предлагаю сразу перейти на react-final-form, я его посмотрел, мне нравится. Очень гибкая штука, апи практически такое же как и у редакс-форм, но расширен фичами по контролю подписок и возможностью расширять функционал форм с помощью различных плагинов.

Единственный кейс который нельзя обработать с помощью файнал форм, это изменить стейт формы извне (а это крайне редкий кейс), поэтому предлагаю в стартер кит взять более легкий и производительный файнал форм, а на конкретном проекте если нужно подключать редакс форм.

@NikitaRzm я думаю тут еще твой аппрув нужен))

Reply:
Если что, вот код этого хока заточенного под файнал форм
```
import * as React from 'react';
import { Field, FieldRenderProps, FieldProps as RFFieldProps } from 'react-final-form';
import { Omit, MergeRight } from '_helpers';

type BaseWrappedFieldProps = FieldRenderProps & {
  value?: any;
  onChange?: any;
};

// keys for pick props from RFFieldProps, Omit dont work because RFFieldProps have index signature
type RFFieldPropKey =
  | 'allowNull' | 'format' | 'formatOnBlur' | 'parse' | 'name'
  | 'isEqual' | 'subscription' | 'validate' | 'value';

function getFieldWithComponent<P extends BaseWrappedFieldProps>(Component: React.ComponentType<P>, type?: string) {
  type OwnProps = Omit<P, keyof BaseWrappedFieldProps>;
  type FieldProps = Pick<RFFieldProps, RFFieldPropKey>;
  type ResultProps = MergeRight<OwnProps, FieldProps>;

  const result: React.StatelessComponent<ResultProps> = (props: ResultProps) =>
    <Field type={type} {...props as any} component={Component} />;
  result.displayName = `FieldWithComponent(${Component.displayName || Component.name || 'Component'})`;
  return result;
}

export default getFieldWithComponent;

```

Reply:
Ок, ща разберусь с redux-final-form и перепилю на них

Reply:
А почему тут деструктурируются пропсы в филд с типом any?
`<Field type={type} {...props as any} component={Component} />;`

Reply:
у меня без any вроде ошибка была, и я не стал тратить время на то чтобы понять из-за чего она)) поэтому заткнул

Reply:
у меня вроде нормас все)

Reply:
ну значит типы поправил сам того не заметив))